src/bsp/pc-linux/build_options.cmake

```diff
@@ -0,0 +1,28 @@
+##########################################################################
+#
+# Build options for "pc-linux" BSP
+#
+##########################################################################
+
+
+
+# Linux system libraries required for the final link of applications using OSAL
+target_link_libraries(osal_bsp
+    pthread dl rt
+)
+
+# C flags that should be used when (re-) compiling code for unit testing.
+# Note: --coverage is just a shortcut for "-ftest-coverage" and "-fprofile-arcs"
+# This also does not work well when cross compiling since paths to the _compile_ dir
+# are baked into the executables, so they will not be there when copied to the target
+# Note - although GCC understands the same flags for compile and link here, this may
+# not be true on all platforms so the compile and link flags are specified separately.
+if (NOT CMAKE_CROSSCOMPILING)
+  set(UT_COVERAGE_COMPILE_FLAGS -pg --coverage)
```

Comment:
This seems to be non-portable:

```
clang: error: the clang compiler does not support -pg option on versions of OS X 10.9 and later
```

Reply:
Another note, even if it might be beyond the scope of this changeset. Usually such functionality is created in separate files like `cmake/coverage.cmake` to not mix the logic of the targets with the logic of the coverage.

The same applies to things like `clang-format`, `clang-tidy`, other code linters, spell checking, etc.

One good example can be found here: https://github.com/StableCoder/cmake-scripts.

Reply:
See #420